tests/test_functional.py

```diff
@@ -97,9 +98,14 @@ def stop_subprocess(self):
             self.proc.terminate()
         self.sock.close()
         # This give us one FD back ...
-        self.queue.close()
         self.proc.join()
         self.proc.close()
+        self.queue.close()
+        self.queue.join_thread()
+
+        import gc
+
+        gc.collect()
```

Comment:
put in some comment here about how you're trying to fix pypy3 specifically for future-folk

Reply:
Done.